src/bloomery/spec/common.py

```
@@ -341,13 +341,168 @@ def validate_document[ModelT: SpecModel](
+#: raw ``RecursionError`` out of ``yaml.load`` (measured). 120 stays far under
+#: the interpreter limit while no hand-written spec nests a tenth of it.
+_MAX_DEPTH = 120
+
+#: Aliases: PyYAML composes an alias as a *shared* node, so a billion-laughs
+#: document loads cheaply here — and then costs whoever walks the result. A
+#: 399-byte document was measured expanding to 10^8 leaves for its first
+#: consumer (validation, in this package). The budget bounds the document's
+#: *expanded* node count — every node counted once per path from the root —
+#: relative to its distinct nodes, so a dense but alias-free document is
+#: never refused for its honest size while a bomb's ratio is astronomical:
+#: legitimate anchor reuse measures well under 2×.
```

Ruff (0.16.3)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)